### PR TITLE
Make imports from `collections` and `collections.abc` consistent

### PR DESCRIPTION
**Pull request by user27182**
### Overview

In #6158, many imports from the `typing` module are deprecated and replaced with imports from `collections.abc` instead. E.g. `from typing import Sequence` -> `from collections.abc import Sequence`.

However, some modules do not import `Sequence` directly into the local namespace, and instead simply do `import collections.abc`. This makes for an inconsistent import scheme across modules.

This PR fixes this and imports from `collections` and `collections.abc` directly into the local namespace to make these imports consistent.

### CODE REVIEW

**Review by tkoyama010 — APPROVED** [review]
LGTM

File: pyvista/core/composite.py [hunk 1]
```diff
@@ -889,7 +889,7 @@ def reverse(self):
         ['sphere', 'cube']
 
         """
-        # Taken from implementation in collections.abc.MutableSequence
+        # Taken from implementation in MutableSequence
```
Comment (user27182) [inline pyvista/core/composite.py:892]: Revert this change
```suggestion
        # Taken from implementation in collections.abc.MutableSequence
```

File: pyvista/core/composite.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ class MultiBlock(
 
     .. versionchanged:: 0.36.0
        ``MultiBlock`` adheres more closely to being list like, and inherits
-       from :class:`collections.abc.MutableSequence`.  Multiple nonconforming
+       from :class:`MutableSequence`.  Multiple nonconforming
```
Comment (user27182) [inline pyvista/core/composite.py:60]: Revert this change
```suggestion
       from :class:`collections.abc.MutableSequence`.  Multiple nonconforming
```

File: pyvista/core/partitioned.py [hunk 3]
```diff
@@ -15,8 +13,11 @@
 from .utilities.helpers import is_pyvista_dataset
 from .utilities.helpers import wrap
 
+if TYPE_CHECKING:
```
Comment (user27182) [inline pyvista/core/partitioned.py:16]:
```suggestion
if TYPE_CHECKING:  # pragma: no cover
```
